pkg/ipam/allocator/multipool/allocator.go

```diff
@@ -23,16 +26,31 @@ type Allocator struct {
 }
 
 func (a *Allocator) Init(ctx context.Context, logger *slog.Logger, _ *metrics.Registry) error {
-	a.poolAlloc = NewPoolAllocator(logger)
+	a.poolAlloc = NewPoolAllocator(logger, nil)
 	a.logger = logger.With(subsysLogAttr...)
 	return nil
 }
 
-func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, _ *metrics.Registry) (allocator.NodeEventHandler, error) {
+func (a *Allocator) Start(ctx context.Context, getterUpdater ipam.CiliumNodeGetterUpdater, reg *metrics.Registry) (allocator.NodeEventHandler, error) {
+	var metricsIPAM metricsAPI
+	if operatorOption.Config.EnableMetrics {
+		triggerMetricsIPAM := ipamMetrics.NewTriggerMetricsIPAM(metrics.Namespace, "remaining_ips")
+		triggerMetricsIPAM.MultipoolRegister(reg)
+		metricsIPAM = triggerMetricsIPAM
+	} else {
+		metricsIPAM = &ipamMetrics.NoOpMetrics{}
+	}
+
+	a.poolAlloc = NewPoolAllocator(a.logger, metricsIPAM)
```

Comment:
Why are we creating a new pool allocator? I wouldn't expect any change in the allocator behavior in the context of this PR